### PR TITLE
Eb doc password

### PR DESCRIPTION
**Pull request by ethanbennett**
- Adds password column with Bcrypt to Repos table
- Doesn't validate for password (optional)
- Uploads allow for password in params
- Downloads authenticate with password only if a repo has a password
- Some (but not all) of Laszlo's most recent branch is in here as well
@jmejia @DavidKnott @Laszlo-JFLMTCO

### CODE REVIEW

File: app/controllers/downloads_controller.rb [hunk 1]
```diff
@@ -2,6 +2,15 @@ class DownloadsController < ApplicationController
   def show
     #.available_repos
     repo = current_user.repos.find(params[:id])
-    send_data Download.from_s3(repo), :filename => repo.name
+    if repo.password.nil? || password_check
+      #instead of test.jpg we will use repo.name
+      send_data Download.from_s3(repo), :filename => "test.jpg"
```
Comment (Laszlo-JFLMTCO) [inline app/controllers/downloads_controller.rb]: @ethanbennett I think this piece has to be modified as `:filename => "test.jpg"` will have our old bug back to downloading each file named `test.jpg`. It should be `:filename => repo.name`

FYI @DavidKnott

Comment (ethanbennett) [reply app/controllers/downloads_controller.rb]: Good call. Just fixed it and pushed up the change

Comment (DavidKnott) [reply app/controllers/downloads_controller.rb]: Good catch Laszlo, I agree :filename => repo.name is the most current versions, but other than that and some issues with the password in the repos I fell okay to merge in an effort to keep moving forward.

File: app/controllers/downloads_controller.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ def show
     repo = current_user.repos.find(params[:id])
     if repo.password.nil? || password_check
       #instead of test.jpg we will use repo.name
-      send_data Download.from_s3(repo), :filename => "test.jpg"
+      send_data Download.from_s3(repo), :filename => repo.name
```
Comment (Laszlo-JFLMTCO) [inline app/controllers/downloads_controller.rb]: @ethanbennett 👍 Thank you for the prompt follow up!!!

**Comment by Laszlo-JFLMTCO** [comment]
@ethanbennett Thank you for getting the password feature implemented for downloads! I am OK to merge as my major concern has been addressed. Thank you.

FYI @DavidKnott

**Pull request merged**